### PR TITLE
auto profiling telemetry tweak

### PR DESCRIPTION
**Pull request by jrieken**
reuse the same code to send the telemetry data for
the renderer and the extension host

send telemetry event without annotated stacktraces
send fake error event with annotated stacktraces

This is for https://github.com/microsoft/vscode/issues/163819

### CODE REVIEW

File: src/vs/platform/profiling/common/profilingTelemetrySpec.ts [hunk 1]
```diff
@@ -3,26 +3,83 @@
...
+		selfTime: sample.selfTime,
+		totalTime: sample.totalTime,
+		percentage: sample.percentage,
+		functionName: sample.location,
+		callstack: sample.caller.map(c => c.location).join('<'),
+		callstackHash,
+		extensionId
+	});
+
+	telemetryService.publicLogError2<ProfilingSampleError, ProfilingSampleErrorClassification>(`${name}.sampleError`, {
```
Comment (jrieken) [inline src/vs/platform/profiling/common/profilingTelemetrySpec.ts:81]: @lramos15 please check if this is correct usage of error telemetry, esp if this goes to the errors page.

Comment (lramos15) [reply src/vs/platform/profiling/common/profilingTelemetrySpec.ts:81]: This won't go to the errors site, but can still be queried in Kusto is that what you want?

**Review by lramos15 — APPROVED** [review]
Telemetry stuff looks good, just the heads up about the error website

File: src/vs/platform/profiling/common/profilingTelemetrySpec.ts (hunk 1 above)
Comment (jrieken) [reply src/vs/platform/profiling/common/profilingTelemetrySpec.ts:81]: Will it have the same size constraints and suffer from data cropping like normal telemetry? I don't want that ;-)

Comment (lramos15) [reply src/vs/platform/profiling/common/profilingTelemetrySpec.ts:81]: That is a pipeline restriction and applies to the error site as well. There's no way around that

**Review by lramos15 — CHANGES_REQUESTED** [review]
Sorry I missed this, we cannot have dynamic event names as it's impossible to classify what is being sent in the catalog

File: src/vs/platform/profiling/common/profilingTelemetrySpec.ts [hunk 2]
```diff
@@ -3,26 +3,60 @@
...
+	const { sample, perfBaseline, source } = data;
+
+	// log a fake error with a clearer stack
+	const fakeError = new Error(`[PerfSampleError]|${sample.selfTime}ms`);
+	fakeError.name = 'PerfSampleError';
+	fakeError.stack = `${fakeError.message} by ${data.source} in ${sample.location}\n` + sample.caller.map(c => `\t at ${c.location} (${c.percentage}%)`).join('\n');
+	logService.error(fakeError);
+
+	// send telemetry event
+	telemetryService.publicLog2<TelemetrySampleData, TelemetrySampleDataClassification>(`unresponsive.sample`, {
```
Comment (lramos15) [inline src/vs/platform/profiling/common/profilingTelemetrySpec.ts:51]: No need for templated string here

**Review by lramos15 — APPROVED** [review]

**Pull request merged**